src/utils/web3.js

```diff
@@ -58,8 +70,29 @@ export function trackXUSDInMetaMask(xusdAddress) {
           address: xusdAddress,
           symbol: 'XUSD',
           decimals: 18,
-          image: 'https://xusd.fi/images/xusd-token-icon.svg',
+          image: 'https://app.xusd.fi/images/currency/xusd-icon-small.svg',
+        },
+      },
+    },
+    console.log
+  )
+}
+
+export function addNetworkInMetaMask(chainId) {
```

Comment:
This should call the generic `wallet_addEthereumChain` instead of metamask-specific version - also rename the function generically and/or just merge it with switchEthereumChain?